File: 0566-reshape-the-matrix/0566-reshape-the-matrix.js
/**
 * @param {number[][]} mat
 * @param {number} r
 * @param {number} c
 * @return {number[][]}
 */
var matrixReshape = function(mat, r, c) {
    let m = mat.length, n = mat[0].length;
    if(m*n != r*c){
        return mat;
    }
    let matrix = new Array(r).fill(0).map(() => new Array(c).fill(0));
    let row = 0, col = 0;
    for(let i=0;i<m;i++){
        for(let j=0;j<n;j++){
            matrix[row][col++] = mat[i][j];
            if(col === c){
                col = 0;
                row++;
            }
        }
    }
    return matrix;
};

File: 0566-reshape-the-matrix/0566-reshape-the-matrix.java
class Solution {
    public int[][] matrixReshape(int[][] mat, int r, int c) {
        int m = mat.length, n = mat[0].length;
        if(m * n != r*c) return mat;
        
        int[][] matrix = new int[r][c];
        int row=0,col=0;
        for(int i=0;i<m;i++){
            for(int j=0;j<n;j++){
                matrix[row][col++] = mat[i][j];
                if(col == c){
                    col = 0;
                    row++;
                }
            }
        }
        return matrix;
    }
}